File: MyMoney.Core.FS/Models/Transaction.fs
namespace MyMoney.Core.FS.Models

open System

type Transaction (date:DateTime, payee:string, category:string, amount:Currency, memo:string) =
    let mutable _Payee = payee
    let mutable _Category = category
    let mutable _Memo = memo
    let mutable _Date = date
    let mutable _Amount = amount

    member this.Payee 
        with get() = _Payee
        and set value = _Payee <- value

    member this.Category
        with get() = _Category
        and set value = _Category <- value

    member this.Amount
        with get() = _Amount
        and set value = _Amount <- value

    member this.Memo
        with get() = _Memo
        and set value = _Memo <- value

    member this.Date
        with get() = _Date
        and set value = _Date <- value

    member this.DateFormatted
        with get() = _Date.ToShortDateString()

    member this.MonthAbbreviated
        with get() = _Date.ToString("MMM")

    member this.AmountFormatted
        with get() = 
            if (_Amount.Value > 0m)
                then "+" + _Amount.ToString()
            elif(_Amount.Value < 0m) then
                let t = Math.Abs(_Amount.Value)
                let tc = Currency(t)
                "-" + tc.ToString()
            else
                "$0.00"

File: MyMoney.Core/Database/DataUpdater/V100ToV110/OldModels/Transaction.cs
using CommunityToolkit.Mvvm.ComponentModel;

namespace MyMoney.DataUpdater.V100ToV110.OldModels
{
    internal partial class Transaction : ObservableObject
    {
        public Transaction(DateTime date, string payee, Core.Models.Category category, Core.Models.Currency amount, string memo)
        {
            Payee = payee;
            Category = category;
            Amount = amount;
            Memo = memo;
            Date = date;

            if (TransactionHash == "")
            {
                // Generate a hash for this transaction
                TransactionHash = Guid.NewGuid().ToString();
            }
        }

        public string TransactionHash { get; private set; } = "";

        [ObservableProperty]
        private string _payee = "";

        [ObservableProperty]
        private Core.Models.Category _category;

        [ObservableProperty]
        private Core.Models.Currency _amount;

        [ObservableProperty]
        private string _memo;

        [ObservableProperty]
        private string _transactionDetail = "";

        [ObservableProperty]
        private DateTime _date;

        public string DateFormatted
        {
            get { return Date.ToShortDateString(); }
        }

        public string MonthAbbreviated
        {
            get { return Date.ToString("MMM"); }
        }

        public string AmountFormatted
        {
            get
            {
                if (Amount.Value > 0m)
                    return "+" + Amount.ToString();
                else if (Amount.Value < 0m)
                    return "-" + new Core.Models.Currency(Math.Abs(Amount.Value)).ToString();
                else
                    return "$0.00";
            }
        }
    }
}
